.github/workflows/labeler.yml

```diff
@@ -0,0 +1,27 @@
+﻿name: Auto Label
+
+on:
+  pull_request_target:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  labeler:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Run Auto Labeler
+      uses: actions/github-script@v7
+      with:
+        script: |
+          const { get_updated_label_set } = await import('${{ github.workspace }}/tools/pull_request_hooks/autoLabel.js');
+          const new_labels = await get_updated_label_set({ github, context });
+          github.rest.issues.setLabels({
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            labels: new_labels,
+          });
+          console.log(`Labels updated: ${new_labels}`);
+        github-token: ${{ steps.app-token-generation.outputs.token || secrets.GITHUB_TOKEN }}
```

Comment:
`github-token` is `secrets.GITHUB_TOKEN` if you leave it unset, that should be enough?

Reply:
I noticed our other actions did it this way, I didn't know if it was boilerplate or made unnecessary thanks to github or what so I just kept it

Reply:
afaik its used because setting up a github app vs. letting github actions taking care of it lets you do more things with everything involving the github API than actions allow. im pretty sure both actions and a github app are able to read and modify issue and PR labels, but if you wanna swap out the action token with an app token then dont forget to include the app token generation step :p